### ISSUE TITLE
Instant crash after installing 1.0.1 Darwin

### ISSUE THREAD

**Issue by ferropop** [HarbourMasters/2ship2harkinian#627]
Instant crash, v1 worked just fine.

**Comment by joshuagrady** [comment]
I'm experiencing the same thing here.

Here's my log.

```
[2024-05-31 21:49:16.426] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:152] [info] Reading archive: /Users/USERNAME/Library/Application Support/com.2ship2harkinian.2s2h/mm.o2r
[2024-05-31 21:49:16.597] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:182] [info] Adding Archive /Users/USERNAME/Library/Application Support/com.2ship2harkinian.2s2h/mm.o2r to Archive Manager
[2024-05-31 21:49:16.602] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:152] [info] Reading archive: /Applications/2s2h.app/Contents/Resources/2ship.o2r
[2024-05-31 21:49:16.604] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:182] [info] Adding Archive /Applications/2s2h.app/Contents/Resources/2ship.o2r to Archive Manager
[21:49:16.769] [os.cpp:27] [error] Failed add SDL game controller mappings from "./gamecontrollerdb.txt" (Invalid RWops)
[21:49:17.002] [gfx_metal.cpp:362] [error] Failed to compile shader library, error program_source:62:11: error: assigning to 'float' from incompatible type 'float3' (vector of 3 'float' values)
texel.w = WRAP(texel.w, -0.51, 1.51);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:57: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
program_source:65:9: error: no matching function for call to 'mod'
texel = WRAP(texel, -0.51, 1.51);texel = clamp(texel, 0.0, 1.0);    if (texel.w < 8.0 / 256.0) discard_fragment();
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:28: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ^~~
program_source:33:8: note: candidate function not viable: no known conversion from 'float4' (vector of 4 'float' values) to 'float3' (vector of 3 'float' values) for 1st argument
float3 mod(float3 a, float3 b) {
       ^
program_source:65:9: error: assigning to 'float4' (vector of 4 'float' values) from incompatible type 'float3' (vector of 3 'float' values)
texel = WRAP(texel, -0.51, 1.51);texel = clamp(texel, 0.0, 1.0);    if (texel.w < 8.0 / 256.0) discard_fragment();
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:57: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~

[2024-05-31 21:49:20.740] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:152] [info] Reading archive: /Users/USERNAME/Library/Application Support/com.2ship2harkinian.2s2h/mm.o2r
[2024-05-31 21:49:20.931] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:182] [info] Adding Archive /Users/USERNAME/Library/Application Support/com.2ship2harkinian.2s2h/mm.o2r to Archive Manager
[2024-05-31 21:49:20.937] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:152] [info] Reading archive: /Applications/2s2h.app/Contents/Resources/2ship.o2r
[2024-05-31 21:49:20.937] [/Users/runner/work/2ship2harkinian/2ship2harkinian/libultraship/src/resource/archive/ArchiveManager.cpp:182] [info] Adding Archive /Applications/2s2h.app/Contents/Resources/2ship.o2r to Archive Manager
[21:49:21.144] [os.cpp:27] [error] Failed add SDL game controller mappings from "./gamecontrollerdb.txt" (Invalid RWops)
[21:49:21.786] [gfx_metal.cpp:362] [error] Failed to compile shader library, error program_source:62:11: error: assigning to 'float' from incompatible type 'float3' (vector of 3 'float' values)
texel.w = WRAP(texel.w, -0.51, 1.51);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:57: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
program_source:65:9: error: no matching function for call to 'mod'
texel = WRAP(texel, -0.51, 1.51);texel = clamp(texel, 0.0, 1.0);    if (texel.w < 8.0 / 256.0) discard_fragment();
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:28: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ^~~
program_source:33:8: note: candidate function not viable: no known conversion from 'float4' (vector of 4 'float' values) to 'float3' (vector of 3 'float' values) for 1st argument
float3 mod(float3 a, float3 b) {
       ^
program_source:65:9: error: assigning to 'float4' (vector of 4 'float' values) from incompatible type 'float3' (vector of 3 'float' values)
texel = WRAP(texel, -0.51, 1.51);texel = clamp(texel, 0.0, 1.0);    if (texel.w < 8.0 / 256.0) discard_fragment();
        ^~~~~~~~~~~~~~~~~~~~~~~~
program_source:36:57: note: expanded from macro 'WRAP'
#define WRAP(x, low, high) mod((x)-(low), (high)-(low)) + (low)
                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~
```

**Comment by garrettjoecox** [comment]
@ferropop @joshuagrady [we have a fix up here](https://github.com/HarbourMasters/2ship2harkinian/pull/628), would appreciate it if you could test and verify it works

Otherwise you can change your rendering backend to OpenGL or revert to version 1.0.0
```diff
"Backend": {
-    "Id": 4,
-    "Name": "Metal"
+    "Id": 3,
+    "Name": "OpenGL"
},
```

**Comment by joshuagrady** [comment]
@garrettjoecox, I can confirm that this build fixed the issue for me. 🙂

**Comment by Hazer** [comment]
@garrettjoecox works for me too.

**Comment by ferropop** [comment]
Worked here as well, thank you!

**Comment by Archez** [comment]
Fix is available in Rika Charlie 1.0.2
